### PR TITLE
Sort by title only when @collections is present

### LINKED ISSUE #1170: Sort by title on collections field should not throw an error if collection is nil.
Apply method sort_by on @collections only when @collections is present.

### PR DESCRIPTION
**Pull request by christinach**
closes #1170

### CODE REVIEW

File: app/views/records/edit_fields/_member_of_collection_ids.html.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <%= f.input :member_of_collection_ids,
-            collection: @collections.sort_by(&:title),
+            collection: (@collections.sort_by(&:title) if @collections.present?),
```
[the remark this answers is not in the document]
Comment (christinach) [reply app/views/records/edit_fields/_member_of_collection_ids.html.erb]: I was thinking about that. Adding a validation for collections.

File: spec/decorators/ephemera_folder_decorator_spec.rb [hunk 2]
```diff
@@ -70,6 +70,14 @@
     end
   end
 
+  context 'Will not error with no collections' do
+    let(:collection) { FactoryBot.create_for_repository(:collection) }
+    let(:resource) { FactoryBot.create_for_repository(:ephemera_folder, member_of_collection_ids: []) }
+    it 'tries to sort by title a nil array' do
+      expect(resource.decorate.collections.sort_by(&:title)).to be_truthy
```
Comment (tpendragon) [inline spec/decorators/ephemera_folder_decorator_spec.rb]: I don't think this test is testing what you expect. The above is pulling the variable @collections set by a controller, not calling collections on a decorator.

**Comment by tpendragon** [comment]
I can't get this to error, and now @christinach can't either, so I'm gonna close it for now.

**Comment by hackmastera** [comment]
Just ran into this; added replication info to the issue.

**Pull request closed without merging**